pushnotifications/src/main/java/com/pusher/pushnotifications/internal/ApplicationLifecycle.kt

```diff
@@ -21,30 +21,47 @@ class AppActivityLifecycleCallbacks: Application.ActivityLifecycleCallbacks {
     var startedCount = 0
     var stoppedCount = 0
     fun appInBackground(): Boolean = startedCount <= stoppedCount
+
+    var currentActivity: WeakReference<Activity>? = null
   }
 
-  override fun onActivityPaused(activity: Activity?) {
+  override fun onActivityPaused(activity: Activity) {
+    Companion.stoppedCount +=1
+    currentActivity = null
   }
 
-  override fun onActivityResumed(activity: Activity?) {
+  override fun onActivityResumed(activity: Activity) {
+    currentActivity = WeakReference(activity)
   }
 
-  override fun onActivityStarted(activity: Activity?) {
+  override fun onActivityStarted(activity: Activity) {
     Companion.startedCount += 1
+    currentActivity = WeakReference(activity)
   }
 
-  override fun onActivityDestroyed(activity: Activity?) = Unit
+  override fun onActivityDestroyed(activity: Activity) {
+    currentActivity?.get()?.let {
+      if (it == activity) { // an activity may get stopped after a new one is resumed
+        currentActivity = null
+      }
+    }
+  }
 
-  override fun onActivitySaveInstanceState(activity: Activity?, outState: Bundle?) {
+  override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle?) {
   }
 
-  override fun onActivityStopped(activity: Activity?) {
+  override fun onActivityStopped(activity: Activity) {
     Companion.stoppedCount +=1
+    currentActivity?.get()?.let {
+      if (it == activity) { // an activity may get stopped after a new one is resumed
+        currentActivity = null
+      }
+    }
```

Comment:
`let` is an homologous of `map`. In this case I woud inline the variable:

``` kotlin
if (currentActivity?.get() == activity) { // an activity may get stopped after a new one is resumed
  currentActivity = null
}
```

or even better, a `when`:

``` kotlin
when(currentActivity?.get()) {
  activity -> currentActivity = null
}
```

We can also do something like this to avoid asymetic flow:
``` kotlin
currentActivity = currentActivity?.takeUnless { it.?get() == activity }
```